sycl/test/basic_tests/vectors/vector_ulonglong.cpp

```diff
@@ -0,0 +1,17 @@
+// RUN: %clangxx -fsycl %s -o %t.out -lOpenCL
+// RUN: %t.out
+
+// Test that vector with 'unsigned long long' elements has enough bits to store
+// value.
+
+#define SYCL_SIMPLE_SWIZZLES
+#include <CL/sycl.hpp>
+
+int main(void) {
+  unsigned long long ref = 1ull - 2ull;
+  auto vec = cl::sycl::vec<unsigned long long, 1>(ref);
+  unsigned long long val = vec.template swizzle<cl::sycl::elem::s0>();
```

Comment:
Should we check the same in the device code?

Reply:
For this particular problem it does not really matter where to check, it can be done either on device or on host. But using host simply reduces test run time, so I have decided to do it on host.

Reply:
I suggest removing `-fsycl` option then (+ `-lsycl`). This will improve test run time even more.

Reply:
Ok.

Reply:
wait, -fsycl makes sense here. This option may change the behavior - it changes the defines for long types, etc.

Reply:
Ok, lets keep it.

Reply:
Hm..., I just thought that keeping -fsycl probably makes sense only when you test it on device.
HOST-only-code should work the same way with and without -fsycl